### PR TITLE
Add notice component

### PR DESCRIPTION
**Pull request by andysellick**
- component replaces notice, consultation notice and withdrawal notices
- component must have a title
- accepts also either a description_text simple string to be wrapped in a para, or description_govspeak, a chunk of HTML to be rendered by govspeak

screen shot 2017-07-27 at 09 24 57

https://trello.com/c/pmyrvkHi/33-1-convert-notice-to-a-component

### CODE REVIEW

**Comment by andysellick** [comment]
Have added a test but it's incomplete, setting to do not merge.

File: test/integration/notice_test.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+
+class NoticeGovspeakTest < ActionDispatch::IntegrationTest
```
Comment (nickcolley) [inline test/integration/notice_test.rb]: I think we should put this in a 'components' directory so it's not confused with the other template level integration tests.

File: test/components/notice_test.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'component_test_helper'
+
+class NoticeTest < ComponentTestCase
+  def component_name
+    "notice"
+  end
+
+  test "fails to render when no parameters are given" do
```
Comment (nickcolley) [inline test/components/notice_test.rb]: We need to figure out what do with these tests, ideally we'd only have the integration test and that'd cover all options.

File: test/integration/notice_test.rb (hunk 1 above)
Comment (fofr) [inline test/integration/notice_test.rb]: Or call it `NoticeComponentGovspeakTest`

Comment (andysellick) [inline test/integration/notice_test.rb]: There's already a notice_test.rb in the test/components/ directory, it contains one test to check that nothing is rendered if no parameters are passed, which the integration test can't check. Could I combine the files? Would that work/be sensible?

File: app/assets/stylesheets/components/_notice.scss [hunk 3]
```diff
@@ -8,7 +8,8 @@
     padding: $gutter-half;
   }
 
-  h2 {
+  // scss-lint:disable SelectorFormat
```
Comment (nickcolley) [inline app/assets/stylesheets/components/_notice.scss]: We can get rid of these selectors now, also the nesting is not necessary anymore

File: test/components/notice_test.rb (hunk 2 above)
Comment (andysellick) [inline test/components/notice_test.rb]: Not sure that's possible - the integration test couldn't test that the component errors if a required parameter isn't passed.

Comment (andysellick) [inline test/components/notice_test.rb]: Current solution is that I've got the integration test housed in the component test - so now there's just one test file. It works, and I've commented the file to explain.

**Comment by nickcolley** [comment]
Good to merge when conflicts are resolved 👍

File: app/assets/stylesheets/components/_notice.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  clear: both;
+  @include responsive-bottom-margin;
+  padding: $gutter-two-thirds;
+  border: 5px solid $govuk-blue;
+
+  @include media(mobile) {
+    padding: $gutter-half;
+  }
+
+  p:last-child {
```
Comment (fofr) [inline app/assets/stylesheets/components/_notice.scss]: As @nickcolley pointed out, this should be explicit that it's styling something within the govspeak component.

**Pull request merged**